new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItConfigDistributionStrategy.java

```diff
@@ -0,0 +1,1148 @@
+    overrideFiles.add(
+        Paths.get(RESOURCE_DIR, "configfiles/configoverridesset1/jdbc-JdbcTestDataSource-0.xml"));
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset1/config.xml"));
+    overrideFiles.add(Paths.get(RESOURCE_DIR, "configfiles/configoverridesset1/version.txt"));
+
+    //create config override map
+    createConfigMapFromFiles(overridecm, overrideFiles, domainNamespace);
+
+    logger.info("patch the domain resource with overridesConfigMap, secrets , cluster and introspectVersion");
+    patchStr
+        = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/overridesConfigMap\", \"value\": \"" + overridecm + "\"},"
+        + "{\"op\": \"add\", \"path\": \"/spec/introspectVersion\", \"value\": \"2\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    verifyIntrospectorRuns();
+    verifyPodsStateNotChanged();
+
+    //wait until config is updated upto 5 minutes
+    withStandardRetryPolicy
+        .conditionEvaluationListener(
+            condition -> logger.info("Waiting for server configuration to be updated"
+                + "(elapsed time {0} ms, remaining time {1} ms)",
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(configUpdated());
+
+    verifyConfigXMLOverride(true);
+    verifyResourceJDBC0Override(true);
+  }
+
+  /**
+   * Test server configuration and JDBC datasource configurations are overridden on restart of pods when
+   * /spec/configuration/overrideDistributionStrategy is set to ON_RESTART.
+   *
+   * <p>Test sets the above field to ON_RESTART and overrides the /spec/configuration/overridesConfigMap and
+   * /spec/configuration/secrets with new configuration and new secrets.
+   *
+   * <p>Verifies after introspector runs the server configuration and JDBC datasource configurations are not
+   * updated. Verifies the overrides are applied only after a domain restart.
+   */
+  @Disabled
+  @Order(3)
+  @Test
+  @DisplayName("Test overrideDistributionStrategy value ON_RESTART")
+  public void testOnRestartOverride() {
+
+    //patching the domain with /spec/configuration/overrideDistributionStrategy: ON_RESTART
+    String patchStr = "["
+        + "{\"op\": \"add\", \"path\": \"/spec/configuration/overrideDistributionStrategy\", "
+        + "\"value\": \"ON_RESTART\"}"
+        + "]";
+    logger.info("Updating domain configuration using patch string: {0}", patchStr);
+    V1Patch patch = new V1Patch(patchStr);
+    assertTrue(patchDomainCustomResource(domainUid, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+        "Failed to patch domain");
+
+    //does changing overrideDistributionStrategy needs restart of server pods?
```

Comment:
add TODO Comments or the BUG/JIRA Number

Reply:
Added a TODO comment